devices/bmp180/bmp180.go

```diff
@@ -0,0 +1,258 @@
+const (
+	No  Oversampling = 0
+	O2x Oversampling = 1
+	O4x Oversampling = 2
+	O8x Oversampling = 3
+)
+
+// Dev is an handle to a bmp180.
+type Dev struct {
+	d  conn.Conn
+	c  calibration
+	os Oversampling
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("BMP180{%s}", d.d)
+}
+
+const (
+	ctrlCommandTemperature = 0x0E
+	ctrlCommandPressure    = 0x14
+)
+
+// bit offsets in regCtrlMeas
+const (
+	ctrlMeasurementControl = 0
+	ctrlStartConversion    = 5
+	ctrlOversampling       = 6
+)
+
+// Sense returns measurements as °C and kPa.
+func (d *Dev) Sense(env *devices.Environment) error {
+	// start conversion for temperature
+	var cmd byte
+	cmd = 1 << ctrlStartConversion
+	cmd |= ctrlCommandTemperature << ctrlMeasurementControl
+
+	if err := d.d.Tx([]byte{regCtrlMeas, cmd}, nil); err != nil {
+		return err
+	}
+
+	time.Sleep(100 * time.Millisecond)
+
+	// read value
+	var tempBuf [2]byte
+	if err := d.readReg(regOutMSB, tempBuf[:]); err != nil {
+		return err
+	}
+
+	UT := uint16(tempBuf[0])<<8 | uint16(tempBuf[1])
+	temp := d.c.compensateTemp(UT)
+
+	// start conversion for pressure
+	cmd = 1 << ctrlStartConversion
+	cmd |= ctrlCommandPressure << ctrlMeasurementControl
+	cmd |= byte(d.os) << ctrlOversampling
+
+	if err := d.d.Tx([]byte{regCtrlMeas, cmd}, nil); err != nil {
+		return err
+	}
+
+	time.Sleep(100 * time.Millisecond)
+
+	// read value
+	var pressureBuf [3]byte
+	if err := d.readReg(regOutMSB, pressureBuf[:]); err != nil {
+		return err
+	}
+
+	UP := (int32(pressureBuf[0])<<16 + int32(pressureBuf[1])<<8 | int32(pressureBuf[2])) >> (8 - d.os)
+
+	pressure := d.c.compensatePressure(UP, int32(UT), d.os)
+
+	env.Temperature = devices.Celsius(temp * 100)
+	env.Pressure = devices.KPascal(pressure)
+	return nil
+}
+
+// Halt is a noop for the BMP180.
+func (d *Dev) Halt() error {
+	return nil
+}
+
+// Opts is optional options to pass to the constructor.
+type Opts struct {
+	Pressure Oversampling
+}
+
+const (
+	bmp180Address = 0x77 // the address of a BMP180 is fixed at 0x77
+	chipID        = 0x55 // contents of the ID register, always 0x55
+)
+
+// New returns an object that communicates over I²C to BMP180 environmental
+// sensor.
+func New(b i2c.Bus, opts *Opts) (*Dev, error) {
+	d := &Dev{
+		d:  &i2c.Dev{Bus: b, Addr: bmp180Address},
+		os: No,
+	}
+
+	if opts != nil {
+		d.os = opts.Pressure
+	}
+
+	if err := d.makeDev(opts); err != nil {
+		return nil, err
+	}
+	return d, nil
+}
+
+var defaults = Opts{
+	Pressure: O4x,
+}
+
+// registers
+const (
+	regChipID           = 0xD0 // register contains the chip id
+	regCalibrationStart = 0xAA // first calibration register address
+	regSoftReset        = 0xE0 // soft reset register
+	regCtrlMeas         = 0xF4 // control measurement
+	regOutMSB           = 0xF6 // MSB of measurement
+	regOutLSB           = 0xF7 // LSB of measurement
+	regOutXLSB          = 0xF8 // extended LSB of measurement
+)
+
+const softResetValue = 0xB6
+
+func (d *Dev) makeDev(opts *Opts) error {
```

Comment:
Since there's no SPI support (the datasheet says it's on custom orders only, so not worth implementing), then there's no need of a makeDev() function.